api/config/utils.go

```diff
@@ -0,0 +1,18 @@
+package config
+
+import (
+	cfgcpi "ocm.software/ocm/api/config/cpi"
+)
+
+// GetConfigured applies config objects of a config context
+// to a configuration struct of type T.
+// A pointer to the configured struct is returned.
+// Attention: T must be a struct type.
+func GetConfigured[T any](ctxp ContextProvider) (*T, error) {
```

Comment:
There is exactly one usage of this method, so I would like to keep it an internal helper